torch/csrc/jit/ir/subgraph_matcher.h

```diff
@@ -45,6 +45,27 @@ struct Match {
  *  - The matcher will not mutate either the pattern graph or the matched graph,
  * but the latter is taken as non-const so that Match may contain non-const
  * pointers.  This enables clients of this API to use Match to drive mutations.
+ *
+ * Note [Multi-output Patterns]
```

Comment:
the non-backtracking algorithm (`O(M*N)` I think: BFS at each node in the graph) or endowing an output ordering (via some kind of topo-sort) might be a good bootcamp task :)

Reply:
Mm, I don't know if such an algorithm exists, maybe I'm missing something? Will be happy to discuss it :)